### ISSUE TITLE
Possible deadlock in ConfigurationStore

### ISSUE THREAD

**Issue by tjwatson** [OpenLiberty/open-liberty#7764]
PR #7551 introduced a possible deadlock that looks like the following:

```
3XMTHREADINFO      "Default Executor-thread-6" J9VMThread:0x0000000002304500, omrthread_t:0x00007FC20C66F150, java/lang/Thread:0x00000000E092CD00, state:B, prio=5
3XMJAVALTHREAD            (java/lang/Thread getId:0x37, isDaemon:true)
3XMTHREADINFO1            (native thread ID:0x32C3, native priority:0x5, native policy:UNKNOWN, vmstate:B, vm thread flags:0x00000281)
3XMTHREADINFO2            (native stack address range from:0x00007FC1C9ED0000, to:0x00007FC1C9F11000, size:0x41000)
3XMCPUTIME               CPU usage total: 1.096510017 secs, current category="Application"
3XMTHREADBLOCK     Blocked on: com/ibm/ws/config/admin/internal/ConfigurationStore@0x00000000E014D668 Owned by: "Config-Thread-10" (J9VMThread:0x0000000001E3E700, java/lang/Thread:0x00000000E0486660)
3XMHEAPALLOC             Heap bytes allocated since last GC cycle=0 (0x0)
3XMTHREADINFO3           Java callstack:
4XESTACKTRACE                at com/ibm/ws/config/admin/internal/ConfigurationStore.save(ConfigurationStore.java:171)
5XESTACKTRACE                   (entered lock: com/ibm/ws/config/admin/internal/ConfigurationStore@0x00000000E014D668, entry count: 1)
4XESTACKTRACE                at com/ibm/ws/config/admin/internal/ExtendedConfigurationImpl.updateCache(ExtendedConfigurationImpl.java:480)
4XESTACKTRACE                at com/ibm/ws/config/xml/internal/ConfigUpdater.updateConfiguration(ConfigUpdater.java:265)
4XESTACKTRACE                at com/ibm/ws/config/xml/internal/ConfigUpdater.updateConfiguration(ConfigUpdater.java:194)
4XESTACKTRACE                at com/ibm/ws/config/xml/internal/ConfigUpdater.update(ConfigUpdater.java:109)
4XESTACKTRACE                at com/ibm/ws/config/xml/internal/ConfigUpdater.update(ConfigUpdater.java:95)
4XESTACKTRACE                at com/ibm/ws/config/xml/internal/ChangeHandler.switchConfiguration(ChangeHandler.java:103)
5XESTACKTRACE                   (entered lock: com/ibm/ws/config/xml/internal/ChangeHandler@0x00000000E014C540, entry count: 1)
4XESTACKTRACE                at com/ibm/ws/config/xml/internal/ConfigRefresher.refreshConfiguration(ConfigRefresher.java:113)
5XESTACKTRACE                   (entered lock: com/ibm/ws/config/xml/internal/ConfigRefresher@0x00000000E014CDC8, entry count: 1)
4XESTACKTRACE                at com/ibm/ws/config/xml/internal/ConfigFileMonitor.onChange(ConfigFileMonitor.java:144)
4XESTACKTRACE                at com/ibm/ws/kernel/filemonitor/internal/MonitorHolder.scheduledScan(MonitorHolder.java:705)
4XESTACKTRACE                at com/ibm/ws/kernel/filemonitor/internal/MonitorHolder.run(MonitorHolder.java:558)
4XESTACKTRACE                at com/ibm/ws/threading/internal/SchedulingRunnableFixedHelper.run(SchedulingRunnableFixedHelper.java:272)

3XMTHREADINFO      "Config-Thread-10" J9VMThread:0x0000000001E3E700, omrthread_t:0x00007FC1D804B390, java/lang/Thread:0x00000000E0486660, state:P, prio=5
3XMJAVALTHREAD            (java/lang/Thread getId:0x20, isDaemon:true)
3XMTHREADINFO1            (native thread ID:0x32AA, native priority:0x5, native policy:UNKNOWN, vmstate:P, vm thread flags:0x00020081)
3XMTHREADINFO2            (native stack address range from:0x00007FC1CAA26000, to:0x00007FC1CAA67000, size:0x41000)
3XMCPUTIME               CPU usage total: 0.128766000 secs, current category="Application"
3XMTHREADBLOCK     Parked on: java/util/concurrent/locks/ReentrantLock$NonfairSync@0x00000000E0F0B680 Owned by: "Default Executor-thread-6" (J9VMThread:0x0000000002304500, java/lang/Thread:0x00000000E092CD00)
3XMHEAPALLOC             Heap bytes allocated since last GC cycle=0 (0x0)
3XMTHREADINFO3           Java callstack:
4XESTACKTRACE                at sun/misc/Unsafe.park(Native Method)
4XESTACKTRACE                at java/util/concurrent/locks/LockSupport.park(LockSupport.java:186)
4XESTACKTRACE                at java/util/concurrent/locks/AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:847)
4XESTACKTRACE                at java/util/concurrent/locks/AbstractQueuedSynchronizer.acquireQueued(AbstractQueuedSynchronizer.java:881)
4XESTACKTRACE                at java/util/concurrent/locks/AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:1210(Compiled Code))
4XESTACKTRACE                at java/util/concurrent/locks/ReentrantLock$NonfairSync.lock(ReentrantLock.java:220(Compiled Code))
4XESTACKTRACE                at java/util/concurrent/locks/ReentrantLock.lock(ReentrantLock.java:296(Compiled Code))
4XESTACKTRACE                at com/ibm/ws/config/admin/internal/ExtendedConfigurationImpl.getReadOnlyProperties(ExtendedConfigurationImpl.java:335)
4XESTACKTRACE                at com/ibm/ws/config/admin/internal/ConfigurationStorageHelper.store(ConfigurationStorageHelper.java:153)
4XESTACKTRACE                at com/ibm/ws/config/admin/internal/ConfigurationStore.saveConfigurationDatas(ConfigurationStore.java:133)
5XESTACKTRACE                   (entered lock: com/ibm/ws/config/admin/internal/ConfigurationStore@0x00000000E014D668, entry count: 1)
4XESTACKTRACE                at com/ibm/ws/config/admin/internal/ConfigurationStore.run(ConfigurationStore.java:180)
```

**Comment by tjwatson** [comment]
Fixed in master.
